src/super_gradients/training/metrics/detection_metrics.py

```diff
@@ -199,16 +203,20 @@ def update(self, preds, target: torch.Tensor, device: str, inputs: torch.tensor,
         setattr(self, self.state_key, accumulated_matching_info + new_matching_info)
 
     def compute(self) -> Dict[str, Union[float, torch.Tensor]]:
-        """Compute the metrics for all the accumulated results.
+        """Compute the metric_outputs for all the accumulated results.
         :return: Metrics of interest
         """
         mean_ap, mean_precision, mean_recall, mean_f1, best_score_threshold = -1.0, -1.0, -1.0, -1.0, -1.0
         accumulated_matching_info = getattr(self, self.state_key)
         best_score_threshold_per_cls = np.zeros(self.num_cls)
         mean_ap_per_class = np.zeros(self.num_cls)
 
+        precision, recall = [], []
+
         if len(accumulated_matching_info):
-            matching_info_tensors = [torch.cat(x, 0) for x in list(zip(*accumulated_matching_info))]
+            import torch
```

Comment:
Import should be probably done on top-level

Reply:
Sure, removed